### DISCUSSION TITLE
Can I integrate Airtable's API with Flowise?

### DISCUSSION THREAD

**Discussion by vonsander** [FlowiseAI/Flowise#30] [Q&A]
Looking to pull market research data from Airtable and analyze it with GPT.

P.S. The Discord link is invalid. Is there a new one available that I can use to join?

**Comment by HenryHengZJ** [comment]
hey @vonsander, feel free join here: https://discord.com/invite/jbaHfsRVBW

**Reply by HenryHengZJ** [reply]
for the question, you can try exporting Airtable data as .csv file and use `CSVLoader` in Flowise to build the flow

**Comment by vonsander** [comment]
Thanks! No way to connect Airtable's API directly though?

https://discord.com/invite/jbaHfsRVBW - Getting the error "Invite Invalid".

**Reply by HenryHengZJ** [reply]
unfortunately for now its not possible. How bout this link: https://discord.gg/e8VTqFB4?

**Comment by jingchang0623-crypto** [comment]
Yes, you can definitely integrate Airtable with Flowise! Here are two approaches:

**Option 1: Using HTTP Request Node**
Flowise has a built-in HTTP Request node that can call any REST API, including Airtable:
1. Add an HTTP Request node
2. Set method to GET/POST
3. Add your Airtable API endpoint: `https://api.airtable.com/v0/YOUR_BASE_ID/YOUR_TABLE_NAME`
4. Add header: `Authorization: Bearer YOUR_AIRTABLE_API_KEY`

**Option 2: Using Custom Component**
You can create a custom n8n node or use the Airtable API directly in a Python function node.

**Pro tip:** For document analysis, you can pull data from Airtable, send it to GPT for analysis, then write results back to Airtable!

Let me know if you need more details! 👍

**Comment by kinthaiofficial** [comment]
Integrating Airtable with Flowise works well via the HTTP Request node or a custom tool. Two approaches:

**Option 1: HTTP Request node** (no code needed)
```
API URL: https://api.airtable.com/v0/{BASE_ID}/{TABLE_NAME}
Method: GET
Headers:
  Authorization: Bearer {AIRTABLE_API_KEY}
  Content-Type: application/json
```

**Option 2: Custom Tool** (more flexible, recommended for agents):

```typescript
// Custom Airtable Tool in Flowise
import { Tool } from "@langchain/core/tools";
import Airtable from "airtable";

class AirtableTool extends Tool {
  name = "airtable_lookup";
  description = "Look up records in Airtable by field value. Input: JSON with {table, field, value}";
  
  async _call(input: string): Promise<string> {
    const { table, field, value } = JSON.parse(input);
    const base = new Airtable({ apiKey: process.env.AIRTABLE_API_KEY })
      .base(process.env.AIRTABLE_BASE_ID!);
    
    const records = await base(table)
      .select({ filterByFormula: `{${field}} = "${value}"`, maxRecords: 10 })
      .firstPage();
      
    return JSON.stringify(records.map(r => r.fields));
  }
}
```

Then in Flowise, add this as a Custom Tool and connect it to your Agent node. The agent can then call it when it needs to look up customer data, inventory, etc.

**Common Airtable + AI use cases**:
- CRM lookup during customer conversations
- Inventory check for e-commerce bots
- Scheduling/booking via Airtable as the data store
- Knowledge base maintained in Airtable, queried by AI

What's the workflow you're trying to build? That'd help suggest whether read-only lookup or read+write is needed.
